chart/openfaas/templates/operator-rbac.yaml

```diff
@@ -64,5 +64,57 @@ subjects:
 - kind: ServiceAccount
   name: {{ .Release.Name }}-operator
   namespace: {{ .Release.Namespace | quote }}
+{{- if .Values.clusterRole}}
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ .Release.Name }}-operator-controller
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    app: {{ template "openfaas.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    component: openaas-operator
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: {{ .Release.Name }}-operator-controller
+  namespace: {{ .Release.Namespace | quote }}
+rules:
+  - apiGroups: ["openfaas.com"]
+    resources: ["functions"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+  - apiGroups: [""]
+    resources: ["services"]
+    verbs: ["get", "list", "watch", "create", "delete", "update"]
+  - apiGroups: ["extensions", "apps"]
+    resources: ["deployments"]
+    verbs: ["get", "list", "watch", "create", "delete", "update"]
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+  - apiGroups: [""]
+    resources: ["pods", "pods/log", "namespaces", "endpoints"]
+    verbs: ["get", "list", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Release.Name }}-operator-controller
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    app: {{ template "openfaas.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    component: openfaas-operator
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
```

Comment:
I wonder if we can reduce the number of rbac objects we need to maintain by having one rbac role for k8s objects and one for crd objects, then we only need two role bindings to give the service account the k8s and crd roles, but we only maintain a single set of ClusterRoles instead of having the cluster roles copy each other

Reply:
I agree duplication is not very good thing in general. I also like the idea of maintaining a single set of roles. I just thought of the `{{- if .Values.operator.create }}` separation. That way  there is almost no chance of messing things up between the two installations. I believe if we merge the roles this the logic would become more "if elsy" in a single file and it would be harder to read. The current approach you this file is for the operator and the other file is for the faas-netes. This is the main reason behind this approach, having said that I can reformat it if requested :+1:

Reply:
Also like the idea of this, feel free to propose it in an issue.